provider/resource_keycloak_openid_client_service_account_role.go

```diff
@@ -60,8 +74,16 @@ func setOpenidClientServiceAccountRoleData(data *schema.ResourceData, serviceAcc
 
 func resourceKeycloakOpenidClientServiceAccountRoleCreate(data *schema.ResourceData, meta interface{}) error {
 	keycloakClient := meta.(*keycloak.KeycloakClient)
-	serviceAccountRole := getOpenidClientServiceAccountRoleFromData(data)
-	err := keycloakClient.NewOpenidClientServiceAccountRole(serviceAccountRole)
+	serviceAccountRole, err := getOpenidClientServiceAccountRoleFromData(data, keycloakClient)
+	if err != nil {
+		return err
+	}
+
+	if serviceAccountRole == nil {
```

Comment:
Can this error ever happen?  I don't think it's possible for `getOpenidClientServiceAccountRoleFromData` to return `nil` without also returning an error.